common/raw_hashtable_metadata_group.h

```diff
@@ -573,12 +625,21 @@ inline auto MetadataGroup::Match(uint8_t tag) const -> MatchRange {
                   "SIMD result '{0}' doesn't match portable result '{1}'",
                   simd_result, portable_result);
   }
-  return UseSIMD ? simd_result : portable_result;
+  // Return whichever result we're using. This uses an invoked lambda to deduce
+  // the type from only the selected return statement, allowing them to be
+  // different types.
+  return [&] {
+    if constexpr (UseSIMD) {
+      return simd_result;
+    } else {
+      return portable_result;
+    }
+  }();
```

Comment:
Given you do this twice, and it's kind of subtle, had you considered a helper function? i.e., something like:

```
// Return whichever result we're using. This uses an invoked lambda to deduce
// the type from only the selected return statement, allowing them to be
// different types.
template <bool If, typename ThenT, typename ElseT>
inline auto ConstexprTernary(ThenT then_val, ElseT else_val) -> auto {
  return [&] {
    if constexpr (If) {
      return then_val;
    } else {
      return else_val;
    }
  }();
}
```

While thinking about this, I also was wondering whether there was a good template solution, which got me thinking about `requires`. So here's that thought:

```
// Behaves as a ternary, but allowing different types on the return.
template <bool If, typename ThenT, typename ElseT> requires (If)
inline auto ConstexprTernary(ThenT then_val, ElseT /*else_val*/) -> ThenT {
  return then_val;
}
template <bool If, typename ThenT, typename ElseT> requires (!If)
inline auto ConstexprTernary(ThenT /*then_val*/, ElseT else_val) -> ElseT {
  return else_val;
}
```

Allowing (either way):

```
return ConstexprTernary<UseSIMD>(simd_result, portable_result);
```

Reply:
Sure, done. Went with `if constexpr` as I try to use that over overloads whenever I can as it seems conceptually simpler. But no need for the nested lambda once its in a template function.